diff --git a/src/point-to-point/model/point-to-point-channel.h b/src/point-to-point/model/point-to-point-channel.h
index 766ccd0..3681cba 100644
--- a/src/point-to-point/model/point-to-point-channel.h
+++ b/src/point-to-point/model/point-to-point-channel.h
@@ -100,13 +100,13 @@ public:
    */
   virtual Ptr<NetDevice> GetDevice (std::size_t i) const;

-protected:
   /**
    * \brief Get the delay associated with this channel
    * \returns Time delay
    */
   Time GetDelay (void) const;

+protected:
   /**
    * \brief Check to make sure the link is initialized
    * \returns true if initialized, asserts otherwise
